tests/runtime/test_grpc_server.py

```diff
@@ -212,6 +213,55 @@ def test_predict_fake_module_error_response(
     assert context.value.code() == grpc.StatusCode.NOT_FOUND
 
 
+def test_rpc_validation_on_predict(
+    sample_task_model_id, runtime_grpc_server, sample_inference_service
+):
+    """Check that the server catches models sent to the wrong task RPCs"""
+    stub = sample_inference_service.stub_class(runtime_grpc_server.make_local_channel())
+    predict_request = sample_inference_service.messages.OtherTaskRequest(
+        sample_inputsampleinputtype=HAPPY_PATH_INPUT
+    )
+    with pytest.raises(grpc.RpcError) as context:
+        stub.OtherTaskPredict(
+            predict_request, metadata=[("mm-model-id", sample_task_model_id)]
```

Comment:
offline discussed changing the fixture to make it more obvious why the RPC here is the wrong RPC